torch/onnx/symbolic.py

```diff
@@ -573,25 +573,37 @@ def symbolic(g, input, all_weights, h0, batch_sizes):
             return _unimplemented("RNN", "batch_first")
         if dropout:
             return _unimplemented("RNN", "dropout")
-        if bidirectional:
-            return _unimplemented("RNN", "bidirectional")
+
+        unidirectional = not bidirectional
 
         prev_output = input
         h_outs = []
 
         sequence_lens = unused(g) if batch_sizes is None else batch_sizes
 
         for i in range(num_layers):
-            weight_ih, weight_hh, bias_ih, bias_hh = all_weights[i]
-
-            bias_concat = g.op('Concat', bias_ih, bias_hh, axis_i=0)
-
-            h_in = h0 if num_layers == 1 else g.op('Slice', h0, axes_i=[0], starts_i=[i], ends_i=[i + 1])
+            if unidirectional:
+                weight_ih, weight_hh, bias_ih, bias_hh = all_weights[i]
+                bias_concat = g.op('Concat', bias_ih, bias_hh, axis_i=0)
+
+                h_in = h0 if num_layers == 1 else g.op('Slice', h0, axes_i=[0], starts_i=[i], ends_i=[i + 1])
+            else:
+                weight_ih = g.op('Concat', all_weights[2 * i][0], all_weights[2 * i + 1][0], axis_i=0)
```

Comment:
in the absence of constant folding it's pretty messy output. But I guess that's the best we can do right now

Reply:
I don't think constant folding would even help - it might collapse like 25% of the slice operations, but there would still be a lot left over. And at least this way you can follow the operations if you really want to.